### ISSUE TITLE
systemd-udevd doesn't read /etc/udev/udev.conf

### ISSUE THREAD

**Issue by evverx** [systemd/systemd#5419]
Before https://github.com/systemd/systemd/pull/5302
```
$ sudo strace -s500 -v -eopen /usr/lib/systemd/systemd-udevd |& grep udev.conf
open("/etc/udev/udev.conf", O_RDONLY|O_CLOEXEC) = 4
open("/etc/udev/udev.conf", O_RDONLY|O_CLOEXEC) = 4
```
After
```
strace -s500 -v -eopen /usr/lib/systemd/systemd-udevd |& grep udev.conf
# nothing
```

**Comment by keszybz** [comment]
c22569e is completely broken: all the Makefile.am additions are missing a `\` line terminator. I'm pretty sure that this wasn't tested at all...

**Comment by poettering** [comment]
@ssahani PLEASE test your PRs before submitting them. I mean, it's definitely my fault that I didn't notice that your PR was obviously borked, but it's absolutely not OK to not even superficially test the changes you submit!

**Comment by ssahani** [comment]
Sorry I missed it .

**Comment by ssahani** [comment]
```
index c89fa93..deb846b 100644
--- i/Makefile.am
+++ w/Makefile.am
 
+systemd_udevd_CFLAGS = \
+       $(AM_CFLAGS) \
+       -D UDEV_PROPAGATE_LOG
+
```
Defining `systemd_udevd_CFLAGS` also not able to open `/etc/udev/udev.conf` . Any idea how to fix it ?

**Comment by evverx** [comment]
> Defining systemd_udevd_CFLAGS also not able to open /etc/udev/udev.conf

`systemd_udevd_SOURCES` doesn't contain `src/libudev/libudev.c`. And `-D UDEV_PROPAGATE_LOG` does nothing.

> Any idea how to fix it ?

Well, `UDEV_PROPAGATE_LOG` should affect `src/libudev/libudev.c`. There are two vars `libudev_la_SOURCES` and `libudev_internal_la_SOURCES`. `libudev_la_SOURCES` affects some tests. `libudev_internal_la_SOURCES` affects everything else. I don't see an easy way to add `-D UDEV_PROPAGATE_LOG`

I'd prefer to revert previous PRs and fix #4525 later (`v234`). I think #4525 is better than "udev.conf has no effect at all"

**Comment by poettering** [comment]
I have now posted a PR reverting both patches, we can look into a proper fix after 233. See #5434.
